File: apps/api/app/schemas/ops.py
from pydantic import BaseModel


class WorkerQueueStatusRead(BaseModel):
    status: str
    pending_ready: int
    pending_delayed: int
    running: int
    failed: int
    completed: int
    active: int


class StatusDetailRead(BaseModel):
    status: str
    name: str
    details: dict[str, str | int | bool | float | None]
    error: str | None = None


class SystemStatusRead(BaseModel):
    status: str
    ocr: StatusDetailRead
    embedding: StatusDetailRead
    qdrant: StatusDetailRead
    worker_queue: WorkerQueueStatusRead

File: apps/web/types/ops.ts
export interface WorkerQueueStatus {
  status: string
  pending_ready: number
  pending_delayed: number
  running: number
  failed: number
  completed: number
  active: number
}

export interface StatusDetail {
  status: string
  name: string
  details: Record<string, string | number | boolean | null>
  error?: string | null
}

export interface SystemStatus {
  status: string
  ocr: StatusDetail
  embedding: StatusDetail
  qdrant: StatusDetail
  llm: StatusDetail
  worker_queue: WorkerQueueStatus
}
